### ISSUE TITLE
gamfit 0.1.123: K=1 circle SAE fails to recover noise-free 1-harmonic on circle in R^32 (in-sample R^2 ~0.28, OOS negative)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#186]
## Summary

`sae_manifold_fit(K=1, d_atom=2, atom_topology='circle')` is in spec the natural model for "one circle embedded in ambient space". Yet on the textbook test case — a single harmonic on the circle embedded in a random 2D plane in R^32 — it reaches only **R^2 ≈ 0.28 in-sample at sigma=0.0 (zero noise)**, and **negative OOS R^2**.

Expected at sigma=0.0 with a model whose hypothesis class exactly matches the data: in-sample R^2 ≈ 1.0.

This is not a noise-robustness gap, it is a fit-quality gap on the noise-free, correctly-specified case. Tested with `n_iter=80` (well past defaults).

Environment: macOS arm64, Python 3.13, gamfit 0.1.123.

## Repro (≤30 lines)

```python
import numpy as np, gamfit

def r2(y, p):
    return 1 - ((y-p)**2).sum() / ((y-y.mean(0))**2).sum()

for sigma in (0.0, 0.05):
    rng = np.random.default_rng(0)
    n, p = 200, 32
    theta = rng.uniform(0, 2*np.pi, n)
    Q, _ = np.linalg.qr(rng.standard_normal((p, 2)))
    X = np.stack([np.cos(theta), np.sin(theta)], 1) @ Q.T
    X = X + sigma * rng.standard_normal(X.shape)
    idx = rng.permutation(n); cut = n//5
    Xte, Xtr = X[idx[:cut]], X[idx[cut:]]
    m = gamfit.sae_manifold_fit(X=Xtr, K=1, d_atom=2, atom_topology="circle",
                                n_iter=80, random_state=0)
    print(f"sigma={sigma}  in_R2={r2(Xtr, np.asarray(m.fitted)):+.3f}  "
          f"oos_R2={r2(Xte, m.predict(Xte)):+.3f}  "
          f"reported={m.reconstruction_r2:+.3f}")
```

## Observed

```
sigma=0.0   in_R2=+0.282  oos_R2=-0.192
sigma=0.05  in_R2=+0.264  oos_R2=-0.249
```

(In the larger n sweep, n in {50,100,500,1000} all stay in the in-sample R^2 ≈ 0.07–0.20 band; never converges to the truth.)

## Possible causes

- Optimization not converging in the IBP / gumbel path even at n_iter=80.
- Periodic basis on circle not aligning with the 2D embedding plane.
- Default `sparsity_weight=1.0` / `smoothness_weight=1.0` actively suppressing the true single atom.

Determinism is fine (same random_state ⇒ bit-identical predictions).

**Comment by SauersML** [comment]
Fixed by `6b247894` (route periodic `atom_dim` into `n_harmonics` while pinning the optimizer `latent_dim=1`) + `81d77b43` (auto-derive `learning_rate=1.0` for softmax / ibp_map).

Root cause for the noise-free K=1 R² ≈ 0.28 collapse: `sae_build_atom_plans` previously set both `latent_dim` and `n_harmonics` to the user-supplied `atom_dim` for Periodic atoms. With `atom_dim ≥ 2`, `latent_dim = 2` forced `build_sae_basis_evaluators` to fall back to the frozen `StaticBasisEvaluator` snapshot — the analytic `PeriodicHarmonicEvaluator` is only used when `latent_dim == 1`. With the basis frozen across Newton steps, neither coords nor logits received gradient updates from a refreshed Phi, and the assignment mass collapsed.

After `6b247894`, periodic atoms always run with `latent_dim=1` (a phase angle) and `n_harmonics = atom_dim`, so the analytic basis evaluator stays live and the harmonic 1-circle hypothesis is exactly representable.

Regression coverage: `sae_manifold_fit_10_steps_one_harmonic_reaches_high_r2` in `src/terms/sae_manifold.rs:2921` asserts in-sample R² ≥ 0.95 in 10 Newton steps on this exact K=1 periodic case.

Closing as fixed.
